ingestion/src/main/java/feast/store/serving/redis/RedisStandaloneIngestionClient.java

```diff
@@ -78,8 +78,11 @@ public boolean isConnected() {
   public void sync() {
     // Wait for some time for futures to complete
     // TODO: should this be configurable?
-    LettuceFutures.awaitAll(60, TimeUnit.SECONDS, futures.toArray(new RedisFuture[0]));
-    futures.clear();
+    try {
+      LettuceFutures.awaitAll(60, TimeUnit.SECONDS, futures.toArray(new RedisFuture[0]));
+    } finally {
+      futures.clear();
```

Comment:
Does this clear values that are already queued up, or will they be added again during retry?

Reply:
they will be added again during the retries. The list holds the future returned by Async Redis commands. In the case of we get an exception from any of the commands after future completion, it will hold the exception, and retries will happen. 